code/modules/mob/living/basic/space_fauna/mushroom.dm

```diff
@@ -0,0 +1,183 @@
+/mob/living/basic/mushroom
+	name = "walking mushroom"
+	desc = "It's a massive mushroom... with legs?"
+	icon_state = "mushroom_color"
+	icon_living = "mushroom_color"
+	icon_dead = "mushroom_dead"
+	mob_biotypes = MOB_ORGANIC | MOB_PLANT
+	butcher_results = list(/obj/item/food/hugemushroomslice = 1)
+
+	response_help_continuous = "pets"
+	response_help_simple = "pet"
+	response_disarm_continuous = "gently pushes aside"
+	response_disarm_simple = "gently push aside"
+	response_harm_continuous = "whacks"
+	response_harm_simple = "whack"
+
+	speed = 1
+	attack_verb_continuous = "chomps"
+	attack_verb_simple = "chomp"
+	attack_sound = 'sound/weapons/bite.ogg'
+	attack_vis_effect = ATTACK_EFFECT_BITE
+
+	faction = list(FACTION_MUSHROOM)
+	speak_emote = list("squeaks")
+	death_message = "fainted!"
+
+	ai_controller = /datum/ai_controller/basic_controller/mushroom
+	var/cap_color = "#ffffff"
+	///Tracks our general strength level gained from eating other shrooms
+	var/powerlevel = 0
+	///If someone tries to cheat the system by attacking a shroom to lower its health, punish them so that it won't award levels to shrooms that eat it
+	var/bruised = FALSE
+	///If we hit three, another mushroom's gonna eat us
+	var/faint_ticker = 0
+	///Where we store our cap icons so we dont generate them constantly to update our icon
+	var/static/mutable_appearance/cap_living
+	///Where we store our cap icons so we dont generate them constantly to update our icon
+	var/static/mutable_appearance/cap_dead
+	///Cooldown that tracks how long its been since revival
+	COOLDOWN_DECLARE(recovery_cooldown)
+
+/mob/living/basic/mushroom/Initialize(mapload)
+	. = ..()
+	melee_damage_lower = rand(4, 6)
+	melee_damage_upper = rand(11,21)
+	maxHealth = rand(50,70)
```

Comment:
also these values are pretty high in comparison to the old mushroom, why is that?

![image](https://github.com/tgstation/tgstation/assets/34697715/5f4684a3-3552-4f5d-913a-c28fea210cfd)

Reply:
Oh I see, you aren't setting the stuff on the typepath and doing the += operation, you're just doing it all in Initialize(). That does work, but it's a strange pattern compared to the rest of basic mob code. I would rather you have melee_damage_lower be set to something like 4 (the average of the rand()), and then we completely overwrite that variable on Initialize() without doing the additive operation.